applications/StructuralMechanicsApplication/CMakeLists.txt

```
@@ -6,168 +6,28 @@ message("**** configuring KratosStructuralMechanicsApplication ****")
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/prestress_membrane_element.cpp
-
-  # Adding elements for adjoint sensitivity analysis
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_finite_difference_shell_element.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_finite_difference_cr_beam_element_3D2N.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_finite_difference_truss_element_3D2N.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_finite_difference_truss_element_linear_3D2N.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_elements/adjoint_solid_element.cpp
-
-  # Solid elements
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/base_solid_element.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/small_displacement.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/axisym_small_displacement.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/total_lagrangian.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/axisym_total_lagrangian.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/updated_lagrangian.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/axisym_updated_lagrangian.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/small_displacement_bbar.cpp
-
-  # Adding the SPRISM element
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/solid_shell_element_sprism_3D6N.cpp
-
-  # Adding the nodal concentrated element
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/nodal_concentrated_element.cpp
-
-  # Adding the spring + damper element
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/spring_damper_element_3D2N.cpp
-
-  ## CONDITIONS
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/base_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/point_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/point_contact_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/axisym_point_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_load_condition_2d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/axisym_line_load_condition_2d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_load_condition_3d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/point_moment_condition_3d.cpp
-
-  # Adding conditions for adjoint sensitivity analysis
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_point_load_condition.cpp
+## StructuralMechanics Core sources
+file(GLOB_RECURSE KRATOS_STRUCTURAL_MECHANICS_APPLICATION_CORE
+    ${CMAKE_CURRENT_SOURCE_DIR}/structural_mechanics_application.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/structural_mechanics_application_variables.cpp
```

Try this

```suggestion
    ${CMAKE_CURRENT_SOURCE_DIR}/structural_mechanics_application_variables.cpp
    ${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/constitutive_law_utilities.cpp
```

To me it looks like this a order problem in the list of compiled files, since cotire makes the unity file using the same order as the list of the sources its adding the template especializations after they are used, and the first time they are used the compiler is creating the definition, so double difinition.

In out CI this doesn't happen becase we use cache instead of cotire, and appveyor uses MSVC so nobody knows what magic is happening there...

---

is it not a problem if I add the same file twice?
=> it is also added later again by the glob_recurse of the `custom_utilities`